extensions/amp-form/0.1/amp-form.js

```diff
@@ -567,12 +571,21 @@ export class AmpForm {
       container.setAttribute('role', 'alert');
       container.setAttribute('aria-labeledby', messageId);
       container.setAttribute('aria-live', 'assertive');
-      return this.templates_.findAndRenderTemplate(container, data)
-          .then(rendered => {
-            rendered.id = messageId;
-            rendered.setAttribute('i-amphtml-rendered', '');
-            container.appendChild(rendered);
-          });
+
+      if (this.templates_.hasTemplate(container)) {
+        this.templates_.findAndRenderTemplate(container, data)
+            .then(rendered => {
+              rendered.id = messageId;
+              rendered.setAttribute('i-amphtml-rendered', '');
+              container.appendChild(rendered);
+            });
+      } else {
+        // TODO(vializ): This will be unnecessary when the AMP Layers
+        // implementation is complete. We call mutateElement here and not
+        // where the template is made visible so that we don't do
+        // redundant layout work when a template is rendered too.
+        this.resources_.mutateElement(container, () => {});
```

Comment:
This is to let AMP know that the AMP elements inside this container are now visible so they get scheduled for layout.

An alternative I considered was adding a new type of template `amp-literal` that just echoes its contents, but that could be 1. confusing for users, since they might equate the word "template" with "value substitution", and 2. it would increase the size of core slightly